packages/openapi-generator/src/generator.ts

```diff
@@ -24,46 +24,15 @@ export async function generate(options: GeneratorOptions): Promise<void> {
     await rmdir(options.outputDir, { recursive: true });
   }
 
-  // TODO: should be recursive
-  const inputFilePaths = (await readdir(options.inputDir)).map(fileName =>
-    resolve(options.inputDir, fileName)
-  );
-
-  inputFilePaths.forEach(async filePath => {
-    const serviceName = parseServiceName(filePath);
-    // TODO: get kebapcase unique directory name
-    const serviceDir = resolve(options.outputDir, serviceName);
-    let openApiDocument;
-    try {
-      openApiDocument = await convertOpenApiSpec(filePath);
-    } catch (err) {
-      logger.error(
-        `Could not convert document at ${filePath} to the format needed for parsing and generation. Skipping service generation.`
-      );
-      return;
-    }
-    const convertedInputFilePath = resolve(serviceDir, 'open-api.json');
-    const parsedOpenApiDocument = await parseOpenApiDocument(
-      openApiDocument,
-      serviceName
-    );
-
-    if (!parsedOpenApiDocument.operations.length) {
-      logger.warn(
-        `The given OpenApi specificaton does not contain any operations. Skipping generation for input file: ${filePath}`
-      );
-      return;
+  if (statSync(options.input).isFile()) {
+    const inputFilePath = options.input;
+    generateFromFile(inputFilePath, options);
+  } else {
+    const inputFilePaths = await recursiveInputPathSearch(options.input);
+    for (const filePath of inputFilePaths) {
+      await generateFromFile(filePath, options);
```

Comment:
I was using this approach before, however when i use `.forEach` instead of the `for( ... of ...)` loop, the files were not generated sequentially, which was an issue because then the `duplicateServiceExists` function wasn't properly working, however as you mentioned below, since I will probably remove the `duplicateServiceExists` function, I guess I should use it again